src/services/redis/routes.ts

```diff
@@ -0,0 +1,49 @@
+import { SECS_PER_5_MINS } from "@config/app"
+import { CouldNotFindError, UnknownRepositoryError } from "@domain/errors"
+import { redis } from "@services/redis"
+
+export const RoutesRepository = (): IRoutesRepository => {
+  const persist = async ({
+    key,
+    routeToCache,
+    time = SECS_PER_5_MINS,
+  }: {
+    key: CacheKey
+    routeToCache: CachedRoute
+    time?: Seconds
+  }): Promise<CachedRoute | RepositoryError> => {
+    try {
+      const value = JSON.stringify(routeToCache)
+      await redis.set(key, value, "EX", time)
+      return routeToCache
+    } catch (err) {
+      return new UnknownRepositoryError(err)
+    }
+  }
+
+  const findByKey = async (key: CacheKey): Promise<CachedRoute | RepositoryError> => {
```

Comment:
if we use Cache/Cached prefixes maybe the repo should be CacheRouteRepository and not RoutesRepository.

Reply:
Hm, potentially because the `CacheKey` would usually be the unique identifier for items in the repository but we construct/handle it differently because we're using redis (implementation detail).

Would be interested in hearing @bodymindarts' take on this whole `RoutesRepository` abstraction

Reply:
Changed a few names and changed the repo to `RoutesCache` ([changes](https://github.com/GaloyMoney/galoy/compare/e184bf8...9f403b0))